common/src/main/java/com/microsoft/identity/common/internal/ui/webview/certbasedauth/NfcSmartcardCertBasedAuthChallengeHandler.java

```diff
@@ -89,11 +119,15 @@ public void onCancel() {
                     public void onCreateConnection() {
                         mDialogHolder.showSmartcardNfcLoadingDialog();
                         if (mCbaManager.isDeviceChanged()) {
-                            //In a future version, an error dialog with a custom message could be shown here instead of a general error.
-                            indicateGeneralException(methodTag, new Exception("Device connected via NFC is different from initially connected device."));
-                            request.cancel();
                             clearPin(pin);
-                            onPausedSmartcardDiscovery();
+                            request.cancel();
+                            prepForNextUserInteraction(new IDisconnectionCallback() {
+                                @Override
+                                public void onClosedConnection() {
+                                    //In a future version, an error dialog with a custom message could be shown here instead of a general error.
+                                    indicateGeneralException(methodTag, new Exception("Device connected via NFC is different from initially connected device."));
```

Comment:
What does indicate do? Do we throw these at some point?

Reply:
indicateGeneralException is a helper method that logs, shows general error dialog, and emits telemetry for when an unexpected exception is thrown.
This method is meant to be used when an exception that I don't have a ton of information about is thrown from YubiKit. My plan was to create more specific error dialogs once the telemetry from customers starts coming in.
For this instance though, I think I should get rid of the call here and instead manually log send telemetry etc.Will update.